Hypermedia.AspNetCore.Siren/HypermediaResourceFilter.cs

```diff
@@ -1,20 +1,27 @@
 ﻿namespace Hypermedia.AspNetCore.Siren
 {
-    using System.Linq;
-    using Entities.Builder;
+    using Hypermedia.AspNetCore.Siren.Resources;
     using Microsoft.AspNetCore.Mvc;
     using Microsoft.AspNetCore.Mvc.Filters;
+    using Microsoft.Extensions.DependencyInjection;
+    using System.Linq;
+    using System.Threading.Tasks;
 
-    internal class HypermediaResourceFilter : IResultFilter
+    internal class HypermediaResourceFilter : IAsyncResultFilter
     {
-        private readonly EntityBuilder _builder;
+        private readonly IServiceScopeFactory _factory;
 
-        public HypermediaResourceFilter(EntityBuilder builder)
+        public HypermediaResourceFilter(IServiceScopeFactory factory)
         {
-            this._builder = builder;
+            this._factory = factory;
         }
 
-        public void OnResultExecuting(ResultExecutingContext context)
+        public Task OnResultExecutingAsync(ResultExecutingContext context)
```

Comment:
We should use the "Async" suffix only async functions.

Reply:
Part of the interface definition, but I will make the implementation async.